### PR TITLE
srb2, srb2kart: refactor

### PR DESCRIPTION
**Pull request by luis-caldas**
###### Description of changes

Removed the WAD patches on both packages, a simple wrap using the argument `SRB2WADDIR` solved the problem.
Added desktop items to both packages.
Moved the data files into their own derivation.
Data files from srb2 were not from the official package, now they are and I've left an option to use the old data files.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by luis-caldas** [comment]
@zeratax

File: pkgs/games/srb2kart/default.nix [hunk 1]
```diff
@@ -12,25 +12,49 @@
...
+, p7zip
+, makeWrapper
+, makeDesktopItem
 }:
 
 let
 
-release_tag = "v1.3";
+  # Common info
+  mainName = "srb2kart";
```
Comment (SuperSandro2000) [inline pkgs/games/srb2kart/default.nix:23]:
```suggestion
```

File: pkgs/games/srb2kart/default.nix [hunk 2]
```diff
@@ -12,25 +12,49 @@
...
+  fixVersion = builtins.replaceStrings ["."] [""] version;
+  description = "SRB2Kart is a classic styled kart racer";
 
-installer = fetchurl {
-  url = "https://github.com/STJr/Kart-Public/releases/download/${release_tag}/srb2kart-v13-Installer.exe";
-  sha256 = "0bk36y7wf6xfdg6j0b8qvk8671hagikzdp5nlfqg478zrj0qf6cs";
-};
+  # Normal assets found on the official release
+  assets = stdenv.mkDerivation rec {
+    pname = "${mainName}-data";
```
Comment (SuperSandro2000) [inline pkgs/games/srb2kart/default.nix:32]:
```suggestion
    pname = "srb2kart-data";
```

File: pkgs/games/srb2kart/default.nix [hunk 3]
```diff
@@ -12,25 +12,49 @@
...
+      mkdir -p $out/share/srb2kart
+      cp -r mdls/ *.kart *.dat *.pdf *.srb $out/share/srb2kart/
+    '';
+  };
 
 in stdenv.mkDerivation rec {
-  pname = "srb2kart";
-  version = "1.3.0";
+
+  pname = mainName;
```
Comment (SuperSandro2000) [inline pkgs/games/srb2kart/default.nix:52]:
```suggestion
  pname = "srb2kart";
```

File: pkgs/games/srb2kart/default.nix [hunk 4]
```diff
@@ -47,39 +71,41 @@ in stdenv.mkDerivation rec {
...
   ];
 
-  patches = [
-    ./wadlocation.patch
-  ];
+  # Desktop icon
+  desktopLink = makeDesktopItem rec {
+    name = "Sonic Robo Blast 2 Kart";
+    exec = mainName;
+    icon = mainName;
```
Comment (SuperSandro2000) [inline pkgs/games/srb2kart/default.nix:88]:
```suggestion
    exec = pname;
    icon = pname;
```

File: pkgs/games/srb2/default.nix [hunk 5]
```diff
@@ -13,29 +13,76 @@
...
+, makeWrapper
+, makeDesktopItem
+, customAssets ? false
 }:
 
 let
 
-assets_version = "2.2.5";
+  # Common info
+  mainName = "srb2";
```
Comment (SuperSandro2000) [inline pkgs/games/srb2/default.nix:25]:
```suggestion
```

File: pkgs/games/srb2/default.nix [hunk 6]
```diff
@@ -13,29 +13,76 @@
...
+  fixVersion = builtins.replaceStrings ["."] [""] version;
+  description = "Sonic Robo Blast 2 is a 3D Sonic the Hedgehog fangame based on a modified version of Doom Legacy";
 
-assets = fetchurl {
-  url = "https://github.com/mazmazz/SRB2/releases/download/SRB2_assets_220/srb2-${assets_version}-assets.7z";
-  sha256 = "1m9xf3vraq9nipsi09cyvvfa4i37gzfxg970rnqfswd86z9v6v00";
-};
+  # Normal assets found on the official release
+  assets = stdenv.mkDerivation rec {
+    pname = "${mainName}-data";
```
Comment (SuperSandro2000) [inline pkgs/games/srb2/default.nix:34]:
```suggestion
    pname = "srb2kart-data";
```

File: pkgs/games/srb2/default.nix [hunk 7]
```diff
@@ -13,29 +13,76 @@
...
 
 let
 
-assets_version = "2.2.5";
+  # Common info
+  mainName = "srb2";
+  version = "2.2.10";
+  owner = "STJr";
+  repo = "SRB2";
+  fixVersion = builtins.replaceStrings ["."] [""] version;
```
Comment (SuperSandro2000) [inline pkgs/games/srb2/default.nix:29]:
```suggestion
```

File: pkgs/games/srb2/default.nix [hunk 8]
```diff
@@ -13,29 +13,76 @@
...
+      mkdir -p $out/share/srb2
+      cp -r * $out/share/srb2
+    '';
+  };
 
 in stdenv.mkDerivation rec {
-  pname = "srb2";
-  version = "2.2.10";
+
+  pname = mainName;
```
Comment (SuperSandro2000) [inline pkgs/games/srb2/default.nix:79]:
```suggestion
  pname = "srb2kart";
```

File: pkgs/games/srb2/default.nix [hunk 9]
```diff
@@ -13,29 +13,76 @@
...
 }:
 
 let
 
-assets_version = "2.2.5";
+  # Common info
+  mainName = "srb2";
+  version = "2.2.10";
+  owner = "STJr";
+  repo = "SRB2";
```
Comment (SuperSandro2000) [inline pkgs/games/srb2/default.nix:28]:
```suggestion
```

File: pkgs/games/srb2/default.nix [hunk 10]
```diff
@@ -13,29 +13,76 @@
...
-  sha256 = "1m9xf3vraq9nipsi09cyvvfa4i37gzfxg970rnqfswd86z9v6v00";
-};
+  # Normal assets found on the official release
+  assets = stdenv.mkDerivation rec {
+    pname = "${mainName}-data";
+    inherit version;
+    nativeBuildInputs = [ unzip ];
+    buildInputs = [ unzip ];
+    src = fetchurl {
+      url = "https://github.com/${owner}/${repo}/releases/download/${repo}_release_${version}/${repo}-v${fixVersion}-Full.zip";
```
Comment (SuperSandro2000) [inline pkgs/games/srb2/default.nix:39]:
```suggestion
      url = "https://github.com/STJr/SRB2/releases/download/SRB2_release_${version}/${repo}-v${lib.replaceStrings ["."] [""] version}-Full.zip";
```

File: pkgs/games/srb2/default.nix [hunk 11]
```diff
@@ -54,40 +101,46 @@ in stdenv.mkDerivation rec {
...
-  postInstall = ''
-    mkdir $out/bin
-    mv $out/lsdlsrb2-${version} $out/bin/srb2
+    cp bin/lsdlsrb2-${version} $out/bin/srb2
+    wrapProgram $out/bin/srb2 --set SRB2WADDIR "${chosenAsset}/share/srb2"
   '';
 
   meta = with lib; {
-    description = "Sonic Robo Blast 2 is a 3D Sonic the Hedgehog fangame based on a modified version of Doom Legacy";
+    inherit description;
```
Comment (SuperSandro2000) [inline pkgs/games/srb2/default.nix:139]:
```suggestion
    description = "Sonic Robo Blast 2 is a 3D Sonic the Hedgehog fangame based on a modified version of Doom Legacy";
```

File: pkgs/games/srb2/default.nix [hunk 12]
```diff
@@ -54,40 +101,46 @@ in stdenv.mkDerivation rec {
...
     homepage = "https://www.srb2.org/";
     platforms = platforms.linux;
     license = licenses.gpl2Plus;
     maintainers = with maintainers; [ zeratax ];
   };
+
```
Comment (SuperSandro2000) [inline pkgs/games/srb2/default.nix:145]:
```suggestion
```

File: pkgs/games/srb2/default.nix [hunk 13]
```diff
@@ -13,29 +13,76 @@
...
-assets = fetchurl {
-  url = "https://github.com/mazmazz/SRB2/releases/download/SRB2_assets_220/srb2-${assets_version}-assets.7z";
-  sha256 = "1m9xf3vraq9nipsi09cyvvfa4i37gzfxg970rnqfswd86z9v6v00";
-};
+  # Normal assets found on the official release
+  assets = stdenv.mkDerivation rec {
+    pname = "${mainName}-data";
+    inherit version;
+    nativeBuildInputs = [ unzip ];
+    buildInputs = [ unzip ];
```
Comment (SuperSandro2000) [inline pkgs/games/srb2/default.nix:37]:
```suggestion

```

File: pkgs/games/srb2/default.nix [hunk 14]
```diff
@@ -13,29 +13,76 @@
...
+      url = "https://github.com/mazmazz/SRB2/releases/download/SRB2_assets_220/srb2-${mVersion}-assets.7z";
+      sha256 = "1m9xf3vraq9nipsi09cyvvfa4i37gzfxg970rnqfswd86z9v6v00";
+    };
+    optional = fetchurl {
+      url = "https://github.com/mazmazz/SRB2/releases/download/SRB2_assets_220/srb2-${mVersion}-optional-assets.7z";
+      sha256 = "1j29jrd0r1k2bb11wyyl6yv9b90s2i6jhrslnh77qkrhrwnwcdz4";
+    };
+  };
+  mazmazzAssets = stdenv.mkDerivation rec {
+    pname = "${mainName}-mazmazz-data";
```
Comment (SuperSandro2000) [inline pkgs/games/srb2/default.nix:62]:
```suggestion
    pname = "srb2-mazmazz-data";
```

File: pkgs/games/srb2/default.nix [hunk 15]
```diff
@@ -13,29 +13,76 @@
...
+    version = mVersion;
+    nativeBuildInputs = [ p7zip ];
+    buildInputs = [ p7zip ];
```
Comment (SuperSandro2000) [inline pkgs/games/srb2/default.nix:65]:
```suggestion
```

File: pkgs/games/srb2/default.nix [hunk 16]
```diff
@@ -13,29 +13,76 @@
...
+      cp -r *pk3 *dta *dat models/ $out/share/srb2/
+    '';
+  };
 
-assets_optional = fetchurl {
-  url = "https://github.com/mazmazz/SRB2/releases/download/SRB2_assets_220/srb2-${assets_version}-optional-assets.7z";
-  sha256 = "1j29jrd0r1k2bb11wyyl6yv9b90s2i6jhrslnh77qkrhrwnwcdz4";
-};
+  # Custom assets from mazmazz
+  mVersion = "2.2.5";
```
Comment (SuperSandro2000) [inline pkgs/games/srb2/default.nix:50]:
```suggestion
  mazmazzVersion = "2.2.5";
```

File: pkgs/games/srb2/default.nix [hunk 17]
```diff
@@ -13,29 +13,76 @@
...
+  mAssets = {
+    main = fetchurl {
+      url = "https://github.com/mazmazz/SRB2/releases/download/SRB2_assets_220/srb2-${mVersion}-assets.7z";
+      sha256 = "1m9xf3vraq9nipsi09cyvvfa4i37gzfxg970rnqfswd86z9v6v00";
+    };
+    optional = fetchurl {
+      url = "https://github.com/mazmazz/SRB2/releases/download/SRB2_assets_220/srb2-${mVersion}-optional-assets.7z";
+      sha256 = "1j29jrd0r1k2bb11wyyl6yv9b90s2i6jhrslnh77qkrhrwnwcdz4";
+    };
+  };
```
Comment (SuperSandro2000) [inline pkgs/games/srb2/default.nix:60]:
```suggestion

```

File: pkgs/games/srb2/default.nix [hunk 18]
```diff
@@ -13,29 +13,76 @@
...
+  mazmazzAssets = stdenv.mkDerivation rec {
+    pname = "${mainName}-mazmazz-data";
+    version = mVersion;
+    nativeBuildInputs = [ p7zip ];
+    buildInputs = [ p7zip ];
+    srcs = [ mAssets.main mAssets.optional ];
```
Comment (SuperSandro2000) [inline pkgs/games/srb2/default.nix:66]:
```suggestion
```
This is unused if unpackPhase is bein overwritten.

File: pkgs/games/srb2/default.nix [hunk 19]
```diff
@@ -13,29 +13,76 @@
...
+
+  pname = mainName;
+  inherit version;
+
+  chosenAsset = if customAssets then mazmazzAssets else assets;
 
   src = fetchFromGitHub {
-    owner = "STJr";
-    repo = "SRB2";
+    inherit owner repo;
```
Comment (SuperSandro2000) [inline pkgs/games/srb2/default.nix:85]: Please revert

File: pkgs/games/srb2/default.nix [hunk 20]
```diff
@@ -54,40 +101,46 @@ in stdenv.mkDerivation rec {
     SDL2
     SDL2_mixer
     zlib
+    makeWrapper
```
Comment (SuperSandro2000) [inline pkgs/games/srb2/default.nix:104]: makeWrapper goes into nativeBuildInputs

File: pkgs/games/srb2/default.nix [hunk 21]
```diff
@@ -54,40 +101,46 @@ in stdenv.mkDerivation rec {
...
   ];
 
-  patches = [
-    ./wadlocation.patch
-  ];
+  # Desktop icon
+  desktopLink = makeDesktopItem rec {
+    name = "Sonic Robo Blast 2";
+    exec = mainName;
+    icon = mainName;
```
Comment (SuperSandro2000) [inline pkgs/games/srb2/default.nix:119]:
```suggestion
    exec = "srb2";
    icon = "srb2";
```

File: pkgs/games/srb2/default.nix [hunk 22]
```diff
@@ -54,40 +101,46 @@ in stdenv.mkDerivation rec {
...
     "-DOPENMPT_INCLUDE_DIR=${libopenmpt.dev}/include"
     "-DSDL2_MIXER_INCLUDE_DIR=${SDL2_mixer}/include/SDL2"
     "-DSDL2_INCLUDE_DIR=${SDL2.dev}/include/SDL2"
   ];
 
-  patches = [
-    ./wadlocation.patch
-  ];
+  # Desktop icon
+  desktopLink = makeDesktopItem rec {
```
Comment (SuperSandro2000) [inline pkgs/games/srb2/default.nix:116]:
```suggestion
  desktopItems = [
    (makeDesktopItem rec {
```

File: pkgs/games/srb2/default.nix [hunk 23]
```diff
@@ -54,40 +101,46 @@ in stdenv.mkDerivation rec {
...
+    mkdir -p $out/bin $out/share $out/share/applications $out/share/pixmaps $out/share/icons
 
-  preConfigure = ''
-    7z x ${assets} -o"/build/source/assets" -aos
-    7z x ${assets_optional} -o"/build/source/assets" -aos
-  '';
+    ln -s ${desktopLink}/share/applications/* $out/share/applications
+
+    cp /build/source/srb2.png $out/share/pixmaps/.
+    cp /build/source/srb2.png $out/share/icons/.
```
Comment (SuperSandro2000) [inline pkgs/games/srb2/default.nix:132]: Please use relative paths.

File: pkgs/games/srb2/default.nix [hunk 24]
```diff
@@ -54,40 +101,46 @@ in stdenv.mkDerivation rec {
...
+    genericName = name;
+    categories = [ "Game" ];
+  };
 
-  postPatch = ''
-    substituteInPlace src/sdl/i_system.c \
-        --replace '@wadlocation@' $out
-  '';
+  installPhase = ''
+    mkdir -p $out/bin $out/share $out/share/applications $out/share/pixmaps $out/share/icons
```
Comment (SuperSandro2000) [inline pkgs/games/srb2/default.nix:127]:
```suggestion
    mkdir -p $out/bin $out/share/applications $out/share/pixmaps $out/share/icons
```

File: pkgs/games/srb2kart/default.nix [hunk 25]
```diff
@@ -12,25 +12,49 @@
...
 let
 
-release_tag = "v1.3";
+  # Common info
+  mainName = "srb2kart";
+  version = "1.3";
+  owner = "STJr";
+  repo = "Kart-Public";
+  fixVersion = builtins.replaceStrings ["."] [""] version;
+  description = "SRB2Kart is a classic styled kart racer";
```
Comment (SuperSandro2000) [inline pkgs/games/srb2kart/default.nix:29]:
```suggestion
```

File: pkgs/games/srb2kart/default.nix [hunk 26]
```diff
@@ -12,25 +12,49 @@
...
-installer = fetchurl {
-  url = "https://github.com/STJr/Kart-Public/releases/download/${release_tag}/srb2kart-v13-Installer.exe";
-  sha256 = "0bk36y7wf6xfdg6j0b8qvk8671hagikzdp5nlfqg478zrj0qf6cs";
-};
+  # Normal assets found on the official release
+  assets = stdenv.mkDerivation rec {
+    pname = "${mainName}-data";
+    inherit version;
+    nativeBuildInputs = [ p7zip ];
+    buildInputs = [ p7zip ];
```
Comment (SuperSandro2000) [inline pkgs/games/srb2kart/default.nix:35]:
```suggestion
```

File: pkgs/games/srb2kart/default.nix [hunk 27]
```diff
@@ -12,25 +12,49 @@
...
+    src = fetchurl {
+      url = "https://github.com/${owner}/${repo}/releases/download/v${version}/srb2kart-v${fixVersion}-Installer.exe";
```
Comment (SuperSandro2000) [inline pkgs/games/srb2kart/default.nix:37]:
```suggestion
      url = "https://github.com/STJr/Kart-Public/releases/download/v${version}/srb2kart-v${lib.replaceStrings ["."] [""] version}-Installer.exe";
```

File: pkgs/games/srb2kart/default.nix [hunk 28]
```diff
@@ -12,25 +12,49 @@
...
+      sha256 = "0bk36y7wf6xfdg6j0b8qvk8671hagikzdp5nlfqg478zrj0qf6cs";
+    };
+    unpackPhase = ''
+      7z -y x "''${src}"
```
Comment (SuperSandro2000) [inline pkgs/games/srb2kart/default.nix:41]:
```suggestion
      7z -y x "${src}"
```

File: pkgs/games/srb2kart/default.nix [hunk 29]
```diff
@@ -47,39 +71,41 @@ in stdenv.mkDerivation rec {
     SDL2
     SDL2_mixer
     zlib
+    makeWrapper
```
Comment (SuperSandro2000) [inline pkgs/games/srb2kart/default.nix:74]: makeWrapper goes into nativeBUildInputs

File: pkgs/games/srb2kart/default.nix [hunk 30]
```diff
@@ -47,39 +71,41 @@ in stdenv.mkDerivation rec {
...
     "-DGME_INCLUDE_DIR=${libgme}/include"
     "-DSDL2_MIXER_INCLUDE_DIR=${SDL2_mixer}/include/SDL2"
     "-DSDL2_INCLUDE_DIR=${SDL2.dev}/include/SDL2"
   ];
 
-  patches = [
-    ./wadlocation.patch
-  ];
+  # Desktop icon
+  desktopLink = makeDesktopItem rec {
```
Comment (SuperSandro2000) [inline pkgs/games/srb2kart/default.nix:85]:
```suggestion
  desktopItems = [
    (makeDesktopItem rec {
```

File: pkgs/games/srb2kart/default.nix [hunk 31]
```diff
@@ -47,39 +71,41 @@ in stdenv.mkDerivation rec {
...
+    genericName = name;
+    categories = [ "Game" ];
+  };
 
-  postPatch = ''
-    substituteInPlace src/sdl/i_system.c \
-        --replace '@wadlocation@' $out
-  '';
+  installPhase = ''
+    mkdir -p $out/bin $out/share $out/share/applications $out/share/pixmaps $out/share/icons
```
Comment (SuperSandro2000) [inline pkgs/games/srb2kart/default.nix:96]:
```suggestion
    mkdir -p $out/bin $out/share/applications $out/share/pixmaps $out/share/icons
```

File: pkgs/games/srb2kart/default.nix [hunk 32]
```diff
@@ -47,39 +71,41 @@ in stdenv.mkDerivation rec {
...
 
-  preConfigure = ''
-    mkdir assets/installer
-    pushd assets/installer
-    unzip ${installer} "*.kart" srb2.srb
-    popd
-  '';
+    ln -s ${desktopLink}/share/applications/* $out/share/applications
```
Comment (SuperSandro2000) [inline pkgs/games/srb2kart/default.nix:98]: Please use copyDesktopItems

File: pkgs/games/srb2kart/default.nix [hunk 33]
```diff
@@ -47,39 +71,41 @@ in stdenv.mkDerivation rec {
...
+
+    cp /build/source/srb2.png $out/share/pixmaps/srb2kart.png
+    cp /build/source/srb2.png $out/share/icons/srb2kart.png
```
Comment (SuperSandro2000) [inline pkgs/games/srb2kart/default.nix:101]: Use relative paths

File: pkgs/games/srb2kart/default.nix [hunk 34]
```diff
@@ -47,39 +71,41 @@ in stdenv.mkDerivation rec {
...
-    mkdir -p $out/bin $out/share/games/SRB2Kart
-    mv $out/srb2kart* $out/bin/
-    mv $out/*.kart $out/share/games/SRB2Kart
+    cp bin/srb2kart-${version} $out/bin/srb2kart
+    wrapProgram $out/bin/srb2kart --set SRB2WADDIR "${assets}/share/srb2kart"
   '';
 
   meta = with lib; {
-    description = "SRB2Kart is a classic styled kart racer";
+    inherit description;
```
Comment (SuperSandro2000) [inline pkgs/games/srb2kart/default.nix:108]:
```suggestion
    description = "SRB2Kart is a classic styled kart racer";
```

File: pkgs/games/srb2kart/default.nix [hunk 35]
```diff
@@ -47,39 +71,41 @@ in stdenv.mkDerivation rec {
...
 
-  patches = [
-    ./wadlocation.patch
-  ];
+  # Desktop icon
+  desktopLink = makeDesktopItem rec {
+    name = "Sonic Robo Blast 2 Kart";
+    exec = mainName;
+    icon = mainName;
+    comment = description;
```
Comment (SuperSandro2000) [inline pkgs/games/srb2kart/default.nix:89]:
```suggestion
    comment = meta.description;
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by pinkcreeper100** [comment]
@luis-caldas are you still interested in this pr?

**Review by zeratax — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
can you please rebase?

**Comment by pinkcreeper100** [comment]
@SuperSandro2000 the author of this pr seems to be inactive apart from on their own repos

**Pull request closed without merging**